students/JerryH/Lesson03/mailroom.py

```diff
@@ -1,63 +1,65 @@
 #!/usr/bin/env python3
 
 #List of sample donors and donations
-donors = [['Bill Gates', 1000],
-['Jeff Bezo', 8976.89],
-['Jane Johnson', 7662.90],
-['Mike Dell', 156374.67],
-['Harry Potter', 726364.23],
-['Bill Gates', 2344.77],
-['Jeff Bezo', 676.89],
-['Jane Johnson', 102.90],
-['Mike Dell', 13784.87],
-['Harry Potter', 7264.09],
-['Jane Johnson', 762.90],
-['Mike Dell', 1374.67],
-['Harry Potter', 72664.23]]
+donors = [
+['Bill Gates', [234.22, 45645.24, 43953.09, 98823]],
+['Jeff Bezo', [4564.23]],
+['Mike Dell', [299.09, 26273.67]],
+['Harry Potter', [8234.09, 83948.04, 7834.23]],
+['Ben Williams', [29283.00, 1334.34]],
+['Guy James', [93.00, 34.34]]
+]
+
+THANK_YOU_OPT, REPORT_OPT, QUIT_OPT = (1, 2, 3)
 
 def prompt():
-    return (int(input("Please choose the following options:\n1) Send a Thank you.\n2) Create a Report.\n3) Quit.\n")))
+    return int(input("Please choose the following options:\n1) Send a Thank you.\n2) Create a Report.\n3) Quit.\n"))
 
 def thank_you_prompt():
-    return (input("Send a Thank You - Please enter a full name or type \"list\" to list the current donors:\n"))
+    return input("Send a Thank You - Please enter a full name or type \"list\" to list the current donors:\n")
 
 def get_all_donor_names(donor_list):
     all_names = []
     for i in range(len(donor_list)):
-        all_names = all_names + [donors[i][0]]
+        all_names.append(donor_list[i][0])
     return all_names
 
-def choice(op = 0):
+def main( op = 0 ):
     while op > 3 or op < 1:
         op = prompt()
 
-        if op == 1: #Append or list donors
+        if op == THANK_YOU_OPT: #Append or list donors
             #Add donor's name or list the exisiting donors
             d_name = thank_you_prompt()
             while d_name.lower() == "list": #provide a list of donors if "list" is entered
                 #Get each unique donor's name from the list
-                for each_name in (sorted(set(get_all_donor_names(donors)))):
+                for each_name in sorted(get_all_donor_names(donors)):
                     print("{}".format(each_name))
                 d_name = thank_you_prompt()
```

Comment:
one of the suggestions from the previous PR didn't save unfortunately, apologies.
again, it's optional, but I'd rather suggest and have you decide on improvements on your own

above could be written in this style:
```
donor_name = None
while not donor_name:
    donor_name = donor_name_prompt()  # more intuitive name ( was thank_you_prompt )
    if donor_name == "list":
        donor_name = None
        list_donor_names()
```
should be a little more readable and less code. Less code is not _always_ good, but often it is 🙂 

while loops are there to repeat the same blocks of code, with the help of condition variable (`donor_name`)
in the code you have this line is repeated `d_name = thank_you_prompt()` which often indicates that while loop logic can be improved. you want to minimize logic branches.